### PR TITLE
Turn off passwword authentication for rsync

### PR DESCRIPTION
**Pull request by alex-deref**
In working with the rsync module I was having trouble with hanging processes, due to password prompts hanging in the background.

### CODE REVIEW

File: files/synchronize.py [hunk 1]
```diff
@@ -173,6 +173,8 @@
    - Expect that dest=~/x will be ~<remote_user>/x even if using sudo.
    - To exclude files and directories from being synchronized, you may add 
      C(.rsync-filter) files to the source directory.
+   - Does not use SSH agent, authentication must be done using keys using a
```
Comment (jody-frankowski) [inline files/synchronize.py:176]: Could you elaborate about the ssh agent? `PasswordAuthentication=no` doesn't prevent the use of the agent.
In any case, ansible should be able to work with an ssh agent. Removing key files passphrases is a security risk for a lot of users.

Comment (alex-deref) [inline files/synchronize.py:176]: I had ssh-agent running, and when I ran ssh, Ansible, or rsync directly on the command-line, all three of them would use it to perform the login.

However, when I used the Ansible synchronize task it would hang, presumably waiting for terminal input. If I added a passphraseless default key, id_rsa, then it was able to successfully login. If I added the PasswordAuthentication=No option (and removed id_rsa key), then it would fail to log in and report an error.

The PasswordAuthentication=no option instructs SSH to not use password authentication, and only use cryptographic key authentication. This is fine because using passwords requires a prompt which caused the hang I observed.

My reason for the note "Does not use SSH agent" was due to the fact that the synchronize module did not discover and use my SSH agent. I hypothesise that it couldn't find the SSH agent because the environment had been modified and connection details weren't available to the subprocess.

**Comment by gregdek** [comment]
@tima Please review this PR to make sure it adheres to the following guidelines:

http://docs.ansible.com/developing_modules.html#module-checklist

If it passes these guidelines, and if you believe it’s a good PR otherwise, please add a comment with "shipit" in the text, and we will flag it for inclusion.

If it fails these guidelines, please add a comment with "needs_revision" in the text, along with the needed changes, and we will flag it for revision.

**Comment by tima** [comment]
This is something I was talking to @abadger and @defionscode about a couple of weeks ago. Using a passphrase does complicate things, but I think it's reasonable enough request to support passphrases in some way. IMO committing this will really irk those people so I'm opposed to it.

**Comment by juliedavila** [comment]
What if we made this an optional parameter?

**Comment by alex-deref** [comment]
How would the password prompt work? Ansible does not relinquish control of the terminal, so that can't be used, and it seems the environment isn't passed through either because it couldn't find my agent, which makes accessing X difficult. It might require changes to Ansible itself.

I'd be happy if it were optional, and would favour the default to be not prompting because I consider a hang to be a poor user experience.

File: files/synchronize.py (hunk 1 above)
Comment (mgedmin) [inline files/synchronize.py:176]: For the record, I've been using Ansible's synchronize module to rsync some files keys from server A to server B, relying on my ssh-agent for authentication.  It worked fine.

(Often what breaks ssh agent auth is using 'sudo' to become root -- sudo clears SSH_AUTH_SOCK from the environment, with the default sudo configuration on Ubuntu.)

**Comment by abadger** [comment]
tl;dr;  The ssh-agent note doesn't seem right but I think we could take the "PasswordAuthentication=no" change for now.

Full explanation:

ssh-agent seems to be working for me.  There could be situations where it works and situations where it  doesn't, though (sudo/su/other become methods might change the environment so that the ssh-agent isn't available.. could be other things that are either limitations or bugs)...  Maybe we need to add the ForwardAgent option explicitly here?

I agree that I can't see a way that password auth for ssh would work easily... might be able to give the module a password parameter and then forward that on to the rsync command.  But I'm not sure that will work without using an expect-like library.  Maybe ssh-pass could be used here in the same way that ssh-pas is used for the main ansible connection.... That's all somewhat complex to add though and I'm pretty sure that password authentication won't work without that being added.  So the disabling of password prompting might be the best method until/unless someone submits a PR to do that.

**Comment by gregdek** [comment]
Just a clarification, sorry @abadger -- what action are you proposing here, exactly?

**Comment by tima** [comment]
If we're just talking about a ssh arg of `PasswordAuthentication=no` couldn't the ssh_args parameter of the synchronize module be used now if this is happening to an end user? Do we see this as a wide-spread problem to embed this setting rather leaving it to the end user? Just thinking outloud.

**Comment by gregdek** [comment]
@tima -- can you help with the status of this versus #2244? 

Just unsure what we're asking of @alex-deref -- need a way to proceed, one way or the other.

**Comment by bcoca** [comment]
closing in favor of feature request https://github.com/ansible/ansible/issues/13099, this will also centralize discussions we've been having across many tickets
